src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderStore.cs

```diff
@@ -0,0 +1,363 @@
+
+        /// <summary>
+        /// Cache of pending <see cref= "ProvenBlockHeader"/> items.
+        /// </summary>
+        /// <remarks>
+        /// Pending <see cref= "ProvenBlockHeader"/> items will be saved to disk every minute.
+        /// </remarks>
+        public ConcurrentDictionary<int, ProvenBlockHeader> PendingBatch { get; }
+
+        /// <summary>
+        /// Store Cache of <see cref= "ProvenBlockHeader"/> items.
+        /// </summary>
+        /// <remarks>
+        /// Items are added to this cache when the caller asks for a <see cref= "ProvenBlockHeader"/>.
+        /// </remarks>
+        public ConcurrentDictionary<int, ProvenBlockHeader> Cache { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the object.
+        /// </summary>
+        /// <param name="chain">Thread safe class representing a chain of headers from genesis</param>
+        /// <param name="dateTimeProvider">Provider of time functions.</param>
+        /// <param name="loggerFactory">Factory to create a logger for this type.</param>
+        /// <param name="provenBlockHeaderRepository">Persistent interface of the <see cref="ProvenBlockHeader"></see> DBreeze repository.</param>
+        /// <param name="nodeLifetime">Allows consumers to perform clean-up during a graceful shutdown.</param>
+        /// <param name="nodeStats">Registers an action used to append node stats when collected.</param>
+        /// <param name="asyncLoopFactory">Factory for creating and also possibly starting application defined tasks inside async loop.</param>
+        public ProvenBlockHeaderStore(
+            ConcurrentChain chain,
+            IDateTimeProvider dateTimeProvider,
+            ILoggerFactory loggerFactory,
+            IProvenBlockHeaderRepository provenBlockHeaderRepository,
+            INodeLifetime nodeLifetime,
+            INodeStats nodeStats,
+            IAsyncLoopFactory asyncLoopFactory)
+        {
+            Guard.NotNull(chain, nameof(chain));
+            Guard.NotNull(loggerFactory, nameof(loggerFactory));
+            Guard.NotNull(provenBlockHeaderRepository, nameof(provenBlockHeaderRepository));
+            Guard.NotNull(nodeLifetime, nameof(nodeLifetime));
+            Guard.NotNull(nodeStats, nameof(nodeStats));
+            Guard.NotNull(asyncLoopFactory, nameof(asyncLoopFactory));
+
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+            this.chain = chain;
+            this.provenBlockHeaderRepository = provenBlockHeaderRepository;
+            this.nodeLifetime = nodeLifetime;
+
+            this.PendingBatch = new ConcurrentDictionary<int, ProvenBlockHeader>();
+            this.Cache = new ConcurrentDictionary<int, ProvenBlockHeader>();
+
+            this.asyncLoopFactory = asyncLoopFactory;
+
+            this.performanceCounter = new BackendPerformanceCounter(dateTimeProvider);
+            nodeStats.RegisterStats(this.AddBenchStats, StatsType.Benchmark, 400);
```

Comment:
DBreeze already has priority of 400. Use 100 for store and 200 for repository for consistency.

Reply:
It's only in store now, and changed to 100.

Going forward maybe we use constants, so we know which priorities are set for which bench stats.